iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/BusFilter.java

```diff
@@ -25,47 +25,56 @@ static BusFilter create(Network n, XMLExportOptions options) {
         Set<String> buses = null;
         if (options.isOnlyMainCc()) {
             buses = new HashSet<>();
-            // keep bus of main cc
-            if (options.getTopologyLevel() == TopologyLevel.BUS_BRANCH) {
-                for (Bus b : n.getBusView().getBuses()) {
-                    if (b.isInMainConnectedComponent()) {
-                        buses.add(b.getId());
-                    }
-                }
-            } else {
-                for (Bus b : n.getBusBreakerView().getBuses()) {
-                    if (b.isInMainConnectedComponent()) {
-                        buses.add(b.getId());
-                    }
-                }
+            addBusOfMainCC(buses, n, options);
+            addBusOfOtherSideOfOpenBranches(buses, n, options);
+
+            if (n.getThreeWindingsTransformerCount() != 0) {
+                throw new AssertionError("TODO");
             }
-            // and also bus at the other side of open branches
-            for (Branch branch : n.getBranches()) {
-                Terminal t1 = branch.getTerminal1();
-                Terminal t2 = branch.getTerminal2();
-                if (options.getTopologyLevel() == TopologyLevel.BUS_BRANCH) {
-                    Bus b1 = t1.getBusView().getConnectableBus();
-                    Bus b2 = t2.getBusView().getConnectableBus();
-                    if ((b1 != null && b1.isInMainConnectedComponent()) && b2 != null && !b2.isInMainConnectedComponent()) {
-                        buses.add(b2.getId());
-                    } else if (b1 != null && !b1.isInMainConnectedComponent() && b2 != null && b2.isInMainConnectedComponent()) {
-                        buses.add(b1.getId());
-                    }
-                } else {
-                    Bus b1 = t1.getBusBreakerView().getConnectableBus();
-                    Bus b2 = t2.getBusBreakerView().getConnectableBus();
-                    if (b1.isInMainConnectedComponent() && !b2.isInMainConnectedComponent()) {
-                        buses.add(b2.getId());
-                    } else if (!b1.isInMainConnectedComponent() && b2.isInMainConnectedComponent()) {
-                        buses.add(b1.getId());
-                    }
+        }
+        return new BusFilter(buses, options);
+    }
+
+    static void addBusOfMainCC(Set<String> buses, Network n, XMLExportOptions options) {
```

Comment:
Seems equivalent to something like this:
```
List<Bus> allBuses = (options.getTopologyLevel() == TopologyLevel.BUS_BRANCH) ? n.getBusView().getBuses() : n.getBusBreakerView().getBuses();
allBuses.stream()
    .filter(b -> b.isInMainConnectedComponent())
    .forEach(b -> buses.add(b.getId()));
```

Reply:
Right but in this PR we should stuck to complexity refactoring and as much possible as possible not refactor the code itself